### ISSUE TITLE
Scaling up a cluster via gsctl

### ISSUE THREAD

**Issue by marians** [giantswarm/gsctl#93]
We'd like to support adding worker nodes to an existing cluster (a. k. a. scaling up).

Removing worker nodes (scaling down) should be implemented as a next step, too. However, for this spec, removing worker nodes is not in scope.

### Details

- For a cluster with all identical worker nodes, the user should not have to specify any worker details. The command should allow the user to express "I want more nodes like the ones I have".
  - For AWS, in the beginning the "more of what I have" way will be the only way to execute the command, as clusters are restricted to identical worker node types.
- If required, the user may specify details on the new nodes required. This should be aligned with [cluster creation](https://docs.giantswarm.io/reference/gsctl/create-cluster/).
- In cluster creation, we apply a default node configuration if the user doesn't specify node settings. When adding nodes, it should basically work the same way.
- Users can add any number of nodes. Probably a maximum node count limit for the cluster may apply.

### Command Naming

With the current grammar of `gsctl <verb> <noun>`, that function would likely become accessible under a command like

```
gsctl edit cluster
gsctl modify cluster
gsctl change cluster
```

In concordance with `kubectl`, we should probably go with the verb `edit`.

**Comment by marians** [comment]
Question: Do we rather want to let the user express how many nodes to add ("add 2 workers") or how many should exist after scaling ("make it so that 10 nodes exist)? The first isn't idempotent, the last is. But idempotency hasn't been a focus of gsctl commands so far.

**Comment by marians** [comment]
Some drafts how the command could work in an idempotent way:

### Idempotent command draft A

Idempotency here means that the command can be executed several times, as it expresses the desired state.

In this form (long form) the user may specify node details.

1. User executes `gsctl edit cluster <cluster_id>`
2. Default editor opens to edit the cluster definition YAML
3. User adds one or several node entries and saves the file
4. gsctl validates the change, shows error info (optional)
5. If valid, gsctl submits the PATCH request to the API

Expected result: Worker nodes with given settings are added to the cluster, so that the cluster has the specified array of workers.

Validation has to ensure that existing workers are not modified or removed.

### Idempotent command draft B

This form of the command would let the user scale the cluster up to the specified number of nodes, adding nodes with default settings.

Given is a cluster with less than 7 worker nodes.

```
gsctl edit cluster <cluster_id> --num-workers 7
```

In this case, no editor would open and the command would be validated and then executed immediately.

Expected result: Worker nodes with default settings are added to the cluster, so that the cluster has 7 worker nodes.

### Idempotent command draft C

Here the user scales the cluster up to a specified size, specifying details of the workers to be added via command line options.

Given is a cluster with less than 7 worker nodes.

#### KVM-based cluster

```
gsctl edit cluster <cluster_id> \
  --num-workers 7 \
  --memory-gb 8 \
  --num-cpus 2 \
  --storage-gb 100
```

#### AWS-based cluster

```
gsctl edit cluster <cluster_id> \
  --num-workers 7 \
  --aws-instance-type m3.xlarge
```

Expected result: Worker nodes with specified settings are added to the cluster, so that the cluster has 7 worker nodes.

**Comment by webwurst** [comment]
How about aligning with `kubectl`?
```
kubectl --namespace thux scale deployment resolver --replicas 3
```

**Comment by marians** [comment]
We discussed this and go with the `gsctl scale` version (idempotent), to possibly keep the `edit` function in a kubectl style for later.

**Comment by marians** [comment]
This is currently blocked by https://github.com/giantswarm/api/issues/433 and https://github.com/giantswarm/api/issues/431

**Comment by marians** [comment]
This is no longer blocked by the API.

The implementation is done in PR https://github.com/giantswarm/gsctl/issues/106 and available for testing.

**Comment by marians** [comment]
https://github.com/giantswarm/gsctl/issues/106 is merged.
